contracts/SpokePool.sol

```diff
@@ -439,17 +439,16 @@ abstract contract SpokePool is
         require(amount <= MAX_TRANSFER_SIZE, "Amount too large");
         require(depositCounter[originToken] <= maxCount, "Above max count");
 
-        // This function assumes that L2 timing cannot be compared accurately and consistently to L1 timing. Therefore,
-        // block.timestamp is different from the L1 EVM's. Therefore, the quoteTimestamp must be within a configurable
-        // buffer of this contract's block time to allow for this variance.
-        // Note also that quoteTimestamp cannot be less than the buffer otherwise the following arithmetic can result
-        // in underflow. This isn't a problem as the deposit will revert, but the error might be unexpected for clients.
+        // Require that quoteTimestamp has a maximum age so that depositors pay an LP fee based on recent HubPool usage.
+        // It is assumed that cross-chain timestamps are normally loosely in-sync, but clock drift can occur. If the
+        // SpokePool time stalls or lags significantly, it is still possible to make deposits by setting quoteTimestamp
+        // within the configured buffer. The owner should pause deposits if this is undesirable. This will underflow if
+        // quoteTimestamp is less than depositQuoteTimeBuffer; this is safe but will throw an unintuitive error.
 
-        //slither-disable-next-line timestamp
+        // slither-disable-next-line timestamp
         require(
-            getCurrentTime() >= quoteTimestamp - depositQuoteTimeBuffer &&
-                getCurrentTime() <= quoteTimestamp + depositQuoteTimeBuffer,
-            "invalid quote time"
+            getCurrentTime() >= quoteTimestamp && getCurrentTime() <= quoteTimestamp + depositQuoteTimeBuffer,
```

Comment:
Potential Refactor:
```suggestion
            quoteTimestamp - getCurrentTime() <= depositQuoteTimeBuffer
```

Reasoning:
The original expression is really looking to check if the quote timestamp exists in the window between `[c_t, c_(t+b)]`, or visually:
```
---------------- c ------------------
       [                    ]
      ct                ct+b

c = quote timestamp
ct = current time
b = deposit time buffer
```

If we apply algebra, we can simplify to:
```
c_t <= c <= c_t + b
0 <= c - c_t <= b
```

Because `getCurrentTime()` and `quoteTimestamp()` are both unsigned ints, I believe any case where `c - c_t` where the answer _should_ be negative will result in an underflow and thus make the resulting value incredibly large. 

As a result, we can leverage this underflow effectively to ignore the lower comparison and we are left with the revised expression.

cc @nicholaspai

Reply:
The buffer is backwards-looking rather than forwards-looking, so `quoteTimestamp` must always be _less_ than `getCurrentTime()` and we're actually verifying that `ct-b <= c <= ct`. But adjusting the change to `getCurrentTime() - quoteTimestamp <= depositQuoteTimeBuffer` works and avoids the 2nd call to `getCurrentTime()`. This saves almost 100 gas on a deposit :+1:

nb. The change to the comment above was also incorrect; there was no more possibility of underflow, but if we apply the above change there will be once more. I think that's probably OK, since we only expect advanced users to end up submitting future timestamps, so they should also be advanced enough to identify where it reverts.